icu4c/source/common/rbbitblb.cpp

```diff
@@ -1392,22 +1397,23 @@ void RBBITableBuilder::exportTable(void *where) {
         RBBIStateDescriptor *sd = (RBBIStateDescriptor *)fDStates->elementAt(state);
         RBBIStateTableRow   *row = (RBBIStateTableRow *)(table->fTableData + state*table->fRowLen);
         if (use8BitsForTable()) {
-            U_ASSERT (-128 < sd->fAccepting && sd->fAccepting <= 127);
-            U_ASSERT (-128 < sd->fLookAhead && sd->fLookAhead <= 127);
-            U_ASSERT (-128 < sd->fTagsIdx && sd->fTagsIdx <= 127);
-            row->r8.fAccepting = (int8_t)sd->fAccepting;
-            row->r8.fLookAhead = (int8_t)sd->fLookAhead;
-            row->r8.fTagIdx    = (int8_t)sd->fTagsIdx;
+            U_ASSERT (sd->fAccepting <= 255);
+            U_ASSERT (sd->fLookAhead <= 255);
+            U_ASSERT (0 <= sd->fTagsIdx && sd->fTagsIdx <= 255);
```

Comment:
The sd->fTagsIdx is now unsigned, so we need not assert non-negative any more, right?

Reply:
ok, makes sense